daemon/cmd/daemon_main.go

```diff
@@ -986,6 +986,10 @@ func init() {
 	flags.Bool(option.EnableCustomCallsName, false, "Enable tail call hooks for custom eBPF programs")
 	option.BindEnv(option.EnableCustomCallsName)
 
+	flags.Bool(option.NetfilterCompatibleMode, false, "If set to true, guarantees that traffic will pass through netfilter and that iptable rules will be enforced. This mode may reduce network throughput. If set to false (default), there are no guarantees: some traffic may skip netfilter.")
+	flags.MarkHidden(option.NetfilterCompatibleMode)
+	option.BindEnv(option.NetfilterCompatibleMode)
```

Comment:
Maybe we should add a check in the agent code to see whether `iptables.KernelHasNetfilter()` is true. Otherwise, the compat mode is not required.

Reply:
I added check to see kernel has netfilter support, before setting `NETFILTER_COMPAT_MODE` Macro.